pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -40,7 +40,7 @@ class SetMethod(PyccelFunction):
         The arguments passed to the function call.
     """
     __slots__ = ('_set_variable',)
-    _attribute_nodes = ('_set_variable',)
+    _attribute_nodes = PyccelFunction._attribute_nodes + ('_set_variable',)
```

Comment:
Can you please explain this change to me?

Reply:
Attribute nodes are explained here: https://github.com/pyccel/pyccel/blob/devel/developer_docs/ast_nodes.md#pyccel-internal-function

and here:
https://github.com/pyccel/pyccel/blob/devel/developer_docs/ast_nodes.md#constructing-a-tree

`__slots__` are special in that they can be set and Python will automatically use the value in the class and the values in the super classes. However this is not true for `_attribute_nodes` as we defined this ourselves.

Without this change `_attribute_nodes` contains `('_set_variable',)` but with the change it contains: `('_args', '_set_variable')`. As `SetMethod` saves its arguments into `self._args` via `PyccelFunction` it is important that `_attribute_nodes` contains `_args` in case we want to use a set method in an inline function